oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/ElasticSearchVersion.java

```diff
@@ -43,7 +43,7 @@ private ElasticSearchVersion(final String distribution, final int major, final i
         this.minor = minor;
 
         if (distribution.equalsIgnoreCase("OpenSearch")) {
-            requestFactory = new V78RequestFactory(this);
+            requestFactory = new V81RequestFactory(this);
             codec = V78Codec.INSTANCE;
```

Comment:
V81 Factory with V78 code?

Reply:
> V81 Factory with V78 code?

`RequestFactory` is to compose the API endpoint, such as `_doc/_update/<id>`, `<index>/_update/<id>`, etc., while `Codec` is to (de)serialize the request body, so they don't have to match in version. For example, they may change the endpoint from `_doc/_update/<id>` to `/_update/<id>` but the (de)serialization of the request body is unchanged.